src/main/kotlin/frc/robot/subsystems/elevator/Elevator.kt

```diff
@@ -0,0 +1,46 @@
+package frc.robot.subsystems.elevator
+
+import edu.wpi.first.units.measure.Distance
+import edu.wpi.first.wpilibj2.command.Command
+import edu.wpi.first.wpilibj2.command.Commands
+import edu.wpi.first.wpilibj2.command.SubsystemBase
+import org.littletonrobotics.junction.Logger
+import java.util.function.DoubleSupplier
+
+class Elevator private constructor(private val io: ElevatorIO) : SubsystemBase() {
+
+    companion object {
+        @Volatile
+        private var instance: Elevator? = null
+
+        fun initialize(io: ElevatorIO) {
+            synchronized(this) {
+                if (Elevator.instance == null) {
+                    instance = Elevator(io)
+                }
+            }
+        }
+        fun getInstance(): Elevator {
+            return Elevator.instance ?: throw IllegalStateException(
+                "Elevator has not been initialized. Call initialize(io: ElevatorIO) first."
+            )
+        }
+    }
+
+    fun setPosition(position: Distance):Command {
+        return Commands.run({io.setHeight(position)},this)
+    }
+
+    fun setPower(percentOutput: DoubleSupplier) :Command{
+        return Commands.run({io.setPower(percentOutput.asDouble)},this)
```

Comment:
Why is it necessary to pass `this`?

Reply:
a45b1819849ab722425722d31cb9a8ff13de7f1f